pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +755,94 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	copyNameNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "name"},
+	)
+	copyFileDiffNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "diff"},
+	)
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileNameToClipboard,
+		OnPress: func() error {
+			name := self.context().GetSelectedItemId()
+			if name == "" {
+				return self.c.ErrorMsg(copyNameNoItemSelectionErr)
+			}
+			if err := self.c.OS().CopyToClipboard(name); err != nil {
+				return self.c.Error(err)
+			}
+			self.c.Toast(self.c.Tr.FileNameCopiedToast)
+			return nil
+		},
+		Key: 'n',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label: self.c.Tr.CopyPathDiffToClipboard,
+		OnPress: func() error {
+			path := self.context().GetSelectedPath()
+			if path == "" {
+				return self.c.ErrorMsg(copyFileDiffNoItemSelectionErr)
+			}
+			diff, err := self.c.Git().Diff.GetPathDiff(path)
```

Comment:
This will copy only the _unstaged changes_?

I tried staging something and then using `y` and `Copy diff from selected path to clipboard`, but it copied only the _unstaged_ changes which was not what I expected. (I don't know what exactly I expected, but definitely not only unstaged changes :) )

And if the entire file is staged, this will _clear_ the clipboard which is not really desired. Either the menu option text could be clarified, or the code is faulty. What would people want to  copy most often, `staged` changes maybe?

Reply:
Mh, good catch @mark2185! I think there's no way to copy both staged and unstaged changes using a specific path (now I get what Jesse was referring to when we discussed this on the issue). Would it make sense to substitute that option with two more specific `Copy unstaged diff from selected path to clipboard` + `Copy staged diff from selected path to clipboard`? Or is it better to just choose one of those? 🤔

Also, I can check if the resulting buffer is empty, and in that case, avoid copying it into the clipboard. Maybe we should notify the user in that case, right?

Reply:
> I think there's no way to copy both staged and unstaged changes using a specific path (now I get what Jesse was referring to when we discussed this on the issue)

`git diff HEAD <file>` gives you both `staged` and `unstaged` changes.

> Or is it better to just choose one of those? 🤔

I'm pretty sure the `Copy staged diff` will be enough, I doubt anyone would want to stage some hunks and then copy _everything else_.

> Also, I can check if the resulting buffer is empty, and in that case, avoid copying it into the clipboard. Maybe we should notify the user in that case, right?

Yep, an alert with "Nothing to copy" would suffice.

Reply:
I think it would make sense to copy the staged changes if the selected path has any, or the unstaged changes if not.

Same for all files: copy the staged changes of all (staged) files if there are any, or the unstaged changes if not.

Also, I don't think it needs to differentiate these in the text of the menu entries: "diff of selected file" and "diff of all files" seems intuitive enough to cover all cases.

BTW, for consistency with the `y` command in the commits panel I would leave out all the redundant "Copy xxx to clipboard" in the menu entries; I'd just make it
```
Title: Copy to clipboard
---
n File name
s Diff of selected file
a Diff of all files
```

Reply:
Thanks, @mark2185 @stefanhaller, for the feedback! The solution described by @stefanhaller keeps everything clean and reasonably straightforward: the new implementation should follow your indications. I'm still not 100% sure that the staged vs unstaged behaviour is intuitive enough not to justify a tooltip, but I'd like to avoid making that UI heavier, either. I think it can be as is.

Reply:
Ah, I didn't consider tooltips. I think it would make sense to add one here. My only question would be if it should be dynamic, i.e. describing what happens in this particular situation based on whether something is staged, or describe in a more general way what the logic is. Maybe the latter is actually better, so that people can learn how it works.

Reply:
I actually agree with keeping it fixed so that the users get use to that pattern. We could end up applying a similar logic elsewhere in the future. I'll add a tooltip with a static content.